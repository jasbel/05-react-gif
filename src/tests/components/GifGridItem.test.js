import React from "react"
import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"

describe('Prueba en <GifGridItem />', () => {

  const title="Un titulo";
  const url = "https://localhost/logo.png"
  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test('Debe de mostrar el componente correctamente', () => {

    expect(wrapper).toMatchSnapshot();
  })

  test('Debe de tener un paarrafo con el titulo', () => {
    const p =wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  })

  test('Debe de tener la imagen igual a la url y alt de los props', () => {
    const img = wrapper.find('img');

    // console.log(img.props().src);

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  })

  test('should de tener animate_fadeIn', () => {
    const div= wrapper.find('div')

    // const divClass = div.props().className.split(' ');
    // expect(divClass[2]).toBe('animate__fadeIn');

    const divClass = div.prop('className');
    expect(divClass.includes('animate__fadeIn')).toBe(true);
  })
  
})

