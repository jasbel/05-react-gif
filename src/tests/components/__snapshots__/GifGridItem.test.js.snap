// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/logo.png"
  />
  <p>
    Un titulo
  </p>
</div>
`;
